src/Trakx.CoinGecko.ApiClient/CoinGeckoClients/CoinGeckoClient.cs

```diff
@@ -0,0 +1,43 @@
+﻿using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Logging;
+using Trakx.Common.DateAndTime;
+using Trakx.Common.Logging;
+
+namespace Trakx.CoinGecko.ApiClient;
+
+public partial class CoinGeckoClient : ICoinGeckoClient
+{
+    internal const string MainQuoteCurrency = Constants.Usd;
+
+    private static readonly TimeSpan DefaultCacheLifeSpan = TimeSpan.FromDays(1);
+    private static readonly ILogger Logger = LoggerProvider.Create<CoinGeckoClient>();
+
+    private readonly IMemoryCache _cache;
+    protected readonly string? _typeName;
+
+    public CoinGeckoClient(
+        IMemoryCache cache,
```

Comment:
[suggestion] I would call the variable memoryCache, as we also have distributed, and now hybrid cache

Reply:
I will rename in another PR.